### DISCUSSION TITLE
Deep linking to individual messages

### DISCUSSION THREAD

**Discussion by powellnorma** [open-webui/open-webui#1779] [General]
Let's say I have a long conversation with an AI about something. In the 13th message it has a particular good and well-written answer/insight. I'd like to be able to reference that message directly in text outside of open-webui, by copy-pasting a link to that particular message. Similar to how one can link to individual answers on a Stackoverflow page.

What do you think?

**Comment by justinh-rahb** [comment]
Best you'll get right now is sharing a link for the chat:
![Screenshot 2024-04-26 at 9 43 23 AM](https://github.com/open-webui/open-webui/assets/52832301/f8ae4070-61d3-4bc3-b13b-3876e5395a3d)
![Screenshot 2024-04-26 at 9 43 31 AM](https://github.com/open-webui/open-webui/assets/52832301/3e8d7f98-64f0-49ea-9487-dbb5d39680cf)

**Comment by powellnorma** [comment]
> Best you'll get right now is sharing a link for the chat

Right.

I think adding this should not be too difficult though, we could add an `id=msg-${i}` to each message div, which could then be used as "anchor".

On site load, we could do something like:
```js
onMount(() => {
  const { hash } = document.location;
  const scrollTo = hash && document.getElementById(hash.slice(1));
  if (scrollTo)
    scrollTo.scrollIntoView();
});
```

See https://stackoverflow.com/a/75379121 for an example

The anchor could be added in Messages.svelte [here](https://github.com/open-webui/open-webui/blob/1092ee9c1c130519be4f17712dda4850165ee2b9/src/lib/components/chat/Messages.svelte#L284)

And the "onMount" stuff probably could be added to c/[id]/+page.svelte [here](https://github.com/open-webui/open-webui/blob/1092ee9c1c130519be4f17712dda4850165ee2b9/src/routes/(app)/c/%5Bid%5D/%2Bpage.svelte#L834)
